### PR TITLE
ci(npm-publish-sim): run tests across linux, mac and windows

### PR DESCRIPTION
**Pull request by LeoMcA**
### Description

Run the kitchensink tests across windows and linux when doing an npm publish simulation.

### Motivation

Avoid bugs like https://github.com/mdn/fred/issues/923, by automatically testing changes to fred within content on Windows.

Running kitchensink tests because my powershell scripting wasn't up to task for keeping the current curl-based tests, and this simplifies the workflow: and opens up the possibility of easily testing things like the writers tools themselves in future.

### Related issues and pull requests

Currently fails, as it should: we have a bug. Will pass when https://github.com/mdn/fred/pull/926 is merged.

### CODE REVIEW

File: .github/workflows/npm-publish-simulation.yml [hunk 1]
```diff
@@ -38,8 +38,23 @@ jobs:
...
+        with:
+          name: tarball
+          path: mdn/fred/${{ steps.build.outputs.tarball }}
+          compression-level: 0 # file already compressed
+
+  test:
+    needs: build
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (caugner) [inline .github/workflows/npm-publish-simulation.yml]: Nit: Slight preference for the following.

```suggestion
        os:
          - ubuntu-latest
          - windows-latest
```

PS: Why not also test on macOS?

File: .github/workflows/npm-publish-simulation.yml [hunk 2]
```diff
@@ -54,41 +69,49 @@ jobs:
...
+      - name: (mdn/content) Download tarball
+        uses: actions/download-artifact@v5
+        with:
+          name: tarball
+
       - name: (mdn/content) Install tarball
         working-directory: mdn/content
-        env:
-          TARBALL: ${{ steps.build.outputs.tarball }}
+        shell: bash
```
Comment (caugner) [inline .github/workflows/npm-publish-simulation.yml:82]: Alternatively, set globally?

If I understand https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#jobsjob_idstepsshell correctly, then `sh` is the default for Linux. Why `bash`? Assuming `bash` is available on Windows, shouldn't `sh` as well?

```yml
defaults:
  run:
    shell: bash
```

File: wdio.conf.js [hunk 3]
```diff
@@ -18,13 +20,13 @@ export const config = {
...
-  baseUrl: "http://localhost:3000/",
+  baseUrl: `http://localhost:${FRED_PORT}/`,
   async before(_, __, browser) {
     console.log("waiting for servers to start");
     await browser.waitUntil(
       async () => {
         try {
-          await browser.url("http://localhost:3000");
+          await browser.url(`http://localhost:${FRED_PORT}`);
           await browser.url("http://localhost:8083");
```
Comment (caugner) [inline wdio.conf.js:30]: Nit: While we're here, also add an env variable for the Playground port?

File: .github/workflows/npm-publish-simulation.yml (hunk 2 above)
Comment (LeoMcA) [reply .github/workflows/npm-publish-simulation.yml:82]: The docs suggest `sh` isn't available on Windows, `bash` is the only `sh`-like shell available there. I only use it in this command because this is the only one which has `$variable` substitution: all other steps I want to run in powershell in windows to catch any possible compatibility errors there.

File: .github/workflows/npm-publish-simulation.yml (hunk 1 above)
Comment (LeoMcA) [reply .github/workflows/npm-publish-simulation.yml]: Yes, I suppose we should test on mac too: my thought was that there shouldn't be many differences between linux/mac, but I'm sure we'll be able to find some obscure edge case bug soon!

File: wdio.conf.js (hunk 3 above)
Comment (LeoMcA) [reply wdio.conf.js:30]: `8083` is rari, and it shouldn't change (unlike fred which has a different default in content/the npm package): we can add one in the future if we need it.
